### PR TITLE
Feature/enable cors

### PR DESCRIPTION
**Pull request by RubenZagon**
Use a global configuration to CORS.

we enable the origin localhost:3000 to comunication with the front in develop, **i don't know if we need another origin in producction**

### CODE REVIEW

File: backend/src/main/java/com/huellapositiva/infrastructure/SecurityConfig.java [hunk 1]
```diff
@@ -23,12 +25,25 @@ PasswordEncoder passwordEncoder() {
         return new BCryptPasswordEncoder();
     }
 
+    @Bean
+    public WebMvcConfigurer corsConfigure(){
+        return new WebMvcConfigurer() {
+            @Override
+            public void addCorsMappings(CorsRegistry registry) {
+                registry.addMapping("/api/v1/**")
+                        .allowedOrigins("http://localhost:3000")
```
Comment (ibaiul) [inline backend/src/main/java/com/huellapositiva/infrastructure/SecurityConfig.java]: Externalize this value to a property that can be set in the different environments or it will not work in DEV and PROD

File: backend/src/main/java/com/huellapositiva/infrastructure/SecurityConfig.java [hunk 2]
```diff
@@ -23,12 +25,25 @@ PasswordEncoder passwordEncoder() {
...
+                        .allowedMethods("GET", "POST", "PUT", "DELETE")
```
Comment (ibaiul) [inline backend/src/main/java/com/huellapositiva/infrastructure/SecurityConfig.java]: Is it worth allowing HEAD and OPTIONS methods? Check if your frontend js framework requires it, I think Angular for instance needed it

File: backend/src/test/java/com/huellapositiva/integration/AwsEmailServiceShould.java [hunk 3]
```diff
@@ -18,7 +18,7 @@
 @ExtendWith(AwsEnvVariablesExtension.class)
 @ExtendWith(LocalstackDockerExtension.class)
 @LocalstackDockerProperties(services = { "ses" })
-@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.NONE, properties = {"huellapositiva.feature.email.enabled=true"})
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT, properties = {"huellapositiva.feature.email.enabled=true"})
```
Comment (ibaiul) [inline backend/src/test/java/com/huellapositiva/integration/AwsEmailServiceShould.java:21]: Why is that necessary?

**Review by ibaiul — CHANGES_REQUESTED** [review]

File: backend/src/test/java/com/huellapositiva/integration/AwsEmailServiceShould.java (hunk 3 above)
Comment (josnih21) [reply backend/src/test/java/com/huellapositiva/integration/AwsEmailServiceShould.java:21]: I was having some issues related with AWS Service Email Test so I figure out why this test has not a defined port, when I change this property with the rest of Cors config enabled this test works again. 

I'm not sure at all why but this change makes test passed with cors config enabled

Comment (josnih21) [reply backend/src/test/java/com/huellapositiva/integration/AwsEmailServiceShould.java:21]: I found this when executing webEnvironment to .NONE throws an exception saying that Its necesary to define this **Consider defining a bean named mvcHandlerMappingIntrospector in your configuration.**
 So that was the reason of changing to DEFINED_PORT

File: backend/src/main/java/com/huellapositiva/infrastructure/SecurityConfig.java (hunk 1 above)
Comment (RubenZagon) [reply backend/src/main/java/com/huellapositiva/infrastructure/SecurityConfig.java]: Okay, done.

File: backend/src/main/java/com/huellapositiva/infrastructure/SecurityConfig.java (hunk 2 above)
Comment (RubenZagon) [reply backend/src/main/java/com/huellapositiva/infrastructure/SecurityConfig.java]: It's not necessary for react, u are right. Remove the allowedMethods

File: backend/src/test/java/com/huellapositiva/integration/AwsEmailServiceShould.java (hunk 3 above)
Comment (ibaiul) [reply backend/src/test/java/com/huellapositiva/integration/AwsEmailServiceShould.java:21]: Miraremos eso en el futuro

**Review by ibaiul — APPROVED** [review]
